torch/distributed/distributed_c10d.py

```diff
@@ -1661,6 +1661,20 @@ def _new_process_group_helper(
     # Note: _new_process_group_helper is only called from init_process_group, which always provides a timeout value
     _check_valid_timeout(timeout)
 
+    # When init_process_group is called with MPI back-end, it does not create any Store objects,
+    # as MPI is fully self-sufficient. As a result the "underlying_store" in PrefixStore is None.
+    # The absence of underlying_store causes inability to create new non-MPI process groups,
+    # as they require a functional Store object. Here, it attempts to create a TCPStore for
+    # non-MPI back-ends for its subsequent usage in new group.
+    if is_initialized() and "mpi" not in backend:
+        default_pg = _get_default_group()
+        if store.underlying_store is None and "mpi" in default_pg.name():
+            rendezvous_iterator = rendezvous("env://", timeout=timeout)
```

Comment:
nit: here we use `init_method` provided by the user. 
https://github.com/pytorch/pytorch/blob/687e5cf8c5a511494c41bf061653e45c3c521bf0/torch/distributed/distributed_c10d.py#L1517-L1522

Is my understanding correct that if user used "mpi" for default group init, they would not have passed an `init_method`? (As you mentioned, MPI is self-contained). And thus, we are hard setting a preferred init_method "env://" here and hopefully user have set it?

Reply:
Yes, I do not think there is any other choice. The `new_group` method does not accept `init_method`. The only alternative is that we store `init_method` used in `init_process_group` (that is not being used for MPI) and re-use it in `new_group`.